Hublsoft.Net.LoginLogout.DataAccess/RegisteredUsersRepository.cs

```diff
@@ -105,5 +105,65 @@ public async Task IncrementFailedLoginAttemptsAsync(int id)
                 await con.CloseAsync();
             }
         }
+
+        public async Task<int> GetFailedLoginAttempsAsync(int id)
+        {
+            if (id == 0)
+            {
+                throw new Exception("Id cannot be 0");
+            }
+
+            var attempCount = 0;
+
+            try
+            {
+                using (var con = new MySqlConnection(ConnectionString))
+                {
+                    await con.OpenAsync();
+
+                    var stm = $"SELECT FailedLoginAttempts FROM RegisteredUsers WHERE Id = {id} ;";
+
+                    using (var cmd = new MySqlCommand(stm, con))
+                    {
+                        attempCount = (int)cmd.ExecuteScalar();
+                    }
+
+                    await con.CloseAsync();
+                }
+            }
+            catch
+            {
+            }
+
+            return attempCount;
+        }
+
+        public async Task LockAccount(int id)
+        {
+            if (id == 0)
+            {
+                throw new ArgumentException(nameof(id));
+            }
+            using (var con = new MySqlConnection(ConnectionString))
+            {
+                await con.OpenAsync();
+
+                var stm = $"UPDATE RegisteredUsers SET Status = 1 WHERE Id = {id} ;";
+
+                using (var cmd = new MySqlCommand(stm, con))
+                {
+                    await cmd.ExecuteNonQueryAsync();
+                }
+
+                await con.CloseAsync();
+
+                var stm2 = $"UPDATE RegisteredUsers SET LockedOutUntilDateTime = '{DateTime.Now.AddDays(1):yyyy-MM-dd hh:mm:ss}' WHERE Id = {id} ;";
```

Comment:
Why have two separate update statements against the same table with the same where clause? These should be combined in to a single update statement, which would give a performance increase because you are only running one command against the database rather than two.

Reply:
Yes agreed. Stored procs would help here.

Reply:
Probably better to use UTC time here